src/Config.js

```diff
@@ -113,50 +115,70 @@ Config.prototype.getLoader = function (ext) {
 };
 
 Config.prototype.initOptions = function () {
-    this.opts = require('nomnom');
+    var self = this;
+
+    this.opts = require('commander');
     this.commands = {};
-    this.commands.serve = this.opts.command('serve').help('Run the server');
-    this.commands.serve.option('path', {
-        position: 1,
-        help: 'Optional project path to load at start.'
-    });
-    this.opts.option('port', {
-        default: 6789,
-        help: 'Port to listen on.'
-    });
-    this.opts.option('host', {
-        default: '127.0.0.1',
-        help: 'Host to listen on.'
-    });
-    this.opts.option('mapnik_version', {
-        full: 'mapnik-version',
-        default: this.defaultMapnikVersion(),
-        help: 'Optional mapnik reference version to be passed to Carto'
-    });
-    this.opts.option('proxy', {
-        help: 'Optional proxy to use when doing http requests'
-    });
-    this.opts.option('keepcache', {
-        full: 'keep-cache',
-        flag: true,
-        help: 'Do not flush cached metatiles on project load'
-    });
-    this.opts.option('renderer', {
-        full: 'renderer',
-        default: 'carto',
-        help: 'Specify a renderer by its name, carto is the default.'
-    });
-    this.opts.option('metatile', {
-        help: 'Override mml metatile setting [Default: mml setting]'
+    this.commands.serve = this.opts.command('serve [path]')
+        .description('Run the server with optional project path to load at start.')
+        .option('--port [port]',
+            'Port to listen on. Default is 6789.',
+            parseInt,
+            6789)
+        .option('--host [host]',
+            'Host to listen on. Defaults to 127.0.0.1.',
+            '127.0.0.1')
+        .option('--proxy <proxy>',
+            'Optional proxy to use when doing http requests.')
+        .option('--mapnik-version [version]',
+            'Optional mapnik reference version to be passed to Carto.',
+            this.defaultMapnikVersion())
+        .option('--keep-cache',
+            'Do not flush cached metatiles on project load.')
+        .option('--renderer [name]',
+            'Specify a renderer by its name, carto is the default.',
+            'carto')
+        .option('--metatile <metatile>',
+            'Override mml metatile setting [Default: mml setting].')
+        .action(function (path, options) {
+            if (_has(options, 'port')) {
```

Comment:
Can't we make this a bit more generic, by looping through properties for example?
Also, I'm not sure we need a new dependency for checking that the values are not "undefined", but I may be missing something.